src/AutoMapper/PropertyMap.cs

```diff
@@ -68,39 +83,53 @@ public MemberInfo SourceMember
         {
             get
             {
-                return _sourceMember ?? GetSourceValueResolvers().OfType<IMemberGetter>().LastOrDefault()?.MemberInfo;
+                return _sourceMember ?? _memberChain.LastOrDefault()?.MemberInfo;
+            }
+            internal set
+            {
+                _sourceMember = value;
+                if (value != null)
+                    SourceType = value.GetMemberType();
             }
-            internal set { _sourceMember = value; }
         }
 
         public bool UseDestinationValue { get; set; }
 
-        internal bool HasCustomValueResolver { get; private set; }
-
         public bool ExplicitExpansion { get; set; }
-
+        public object CustomValue { get; private set; }
         public object NullSubstitute { get; private set; }
+        public ValueResolverConfiguration ValueResolverConfig { get; set; }
+        public string SourceMemberName { get; set; }
 
-        public IEnumerable<IValueResolver> GetSourceValueResolvers()
+        public void ChainMembers(IEnumerable<IMemberGetter> members)
         {
-            if (_customMemberResolver != null)
-                yield return _customMemberResolver;
-
-            if (_customResolver != null)
-                yield return _customResolver;
-
-            foreach (var resolver in _sourceValueResolvers)
-            {
-                yield return resolver;
-            }
-
-            if (NullSubstitute != null)
-                yield return new NullReplacementMethod(NullSubstitute);
+            var getters = members as IList<IMemberGetter> ?? members.ToList();
+            _memberChain.AddRange(getters);
+            SourceType = getters.LastOrDefault()?.MemberType;
         }
 
-        public void RemoveLastResolver()
+        public void ApplyInheritedPropertyMap(PropertyMap inheritedMappedProperty)
         {
-            _sourceValueResolvers.RemoveLast();
+            if (!CanResolveValue() && inheritedMappedProperty.IsIgnored())
+            {
+                Ignore();
+            }
+            CustomExpression = CustomExpression ?? inheritedMappedProperty.CustomExpression;
+            _customResolverFunc = _customResolverFunc ?? inheritedMappedProperty._customResolverFunc;
+            if (_condition == null && inheritedMappedProperty._condition != null)
+            {
+                ApplyCondition(inheritedMappedProperty._condition);
+            }
+            if (NullSubstitute == null)
+            {
+                SetNullSubstitute(inheritedMappedProperty.NullSubstitute);
+            }
+            if (_mappingOrder == 0)
+            {
+                SetMappingOrder(inheritedMappedProperty._mappingOrder);
+            }
+            SourceType = SourceType ?? inheritedMappedProperty.SourceType;
+            CustomValue = CustomValue ?? inheritedMappedProperty.CustomValue;

```

Comment:
The idea is I don't want to overwrite something already set. This would be easier if the inherited property map got applied BEFORE the property map stuff. Which I could probably do eventually.